### ISSUE TITLE
Problems with state component: State not found

### ISSUE THREAD

**Issue by WebdevAlexTV** [replit/kaboom#406]
Hey, i have a problem with the relatively new state component. Here is my player initialization with a call afterwards to enter the idle state.

```
const player = k.add([
        k.pos(16, 16),
        k.sprite("player", { frame: 0 }),
        k.solid(),
        k.area(),
        k.body(),
        k.origin("center"),
        k.state("idle", ["idle", "run", "shovel", "pray"]),
        "player"
    ]);

    ...

    player.enterState("idle");
```

This call resolves into the following error:

```
Uncaught Error: State not found: idle
    at Object.enterState (webpack://beneath-unholy-ground/./node_modules/kaboom/dist/kaboom.mjs?:59)
    at initPlayer (webpack://beneath-unholy-ground/./src/player.js?:37)
    at Object.game (webpack://beneath-unholy-ground/./src/scenes/game.js?:39)
    at eval (webpack://beneath-unholy-ground/./node_modules/kaboom/dist/kaboom.mjs?:59)
    at eval (webpack://beneath-unholy-ground/./node_modules/kaboom/dist/kaboom.mjs?:59)
    at Map.forEach (<anonymous>)
    at Object.trigger (webpack://beneath-unholy-ground/./node_modules/kaboom/dist/kaboom.mjs?:59)
    at updateFrame (webpack://beneath-unholy-ground/./node_modules/kaboom/dist/kaboom.mjs?:59)
    at eval (webpack://beneath-unholy-ground/./node_modules/kaboom/dist/kaboom.mjs?:60)
    at frame (webpack://beneath-unholy-ground/./node_modules/kaboom/dist/kaboom.mjs?:59)
```

The version i have installed:

```
2000.1.0
```

Kind regards :)

**Comment by micheg** [comment]
And I don't think it will ever work, not with the current code.
The states are an array of strings, but the comparison is done as if it were a dictionary:

	enterState(state: string, ...args) {
		if (stateList && !stateList[state]) {
			throw new Error(`State not found: ${state}`);
		}

I believe the code should be:

	if (stateList && !stateList.includes(state)) {
		throw new Error(`State not found: ${state}`);
	}

which seems more sensible to me.

**Comment by slmjkdbtl** [comment]
Thanks to @micheg fixed in #408. Will be released in a patch

**Comment by WebdevAlexTV** [comment]
I have installed kaboom 2000.1.2. But now i'm running into another error.

```
Uncaught TypeError: Cannot read properties of undefined (reading 'leave')
    at Object.enterState (kaboom.mjs?:formatted:3645)
    at initPlayer (player.js:37)
    at Object.game (game.js:39)
    at eval (kaboom.mjs?:formatted:3731)
    at eval (kaboom.mjs?:formatted:2378)
    at Map.forEach (<anonymous>)
    at Object.trigger (kaboom.mjs?:formatted:2378)
    at updateFrame (kaboom.mjs?:formatted:4108)
    at eval (kaboom.mjs?:formatted:4312)
    at frame (kaboom.mjs?:formatted:1681)
```

**Comment by slmjkdbtl** [comment]
@WebdevAlexTV It's a bug, fixed in aae0177 and released in `v2000.1.4`.
